lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/IMRUDriver.cs

```diff
@@ -77,6 +77,7 @@ internal sealed class IMRUDriver<TMapInput, TMapOutput, TResult> : IObserver<IDr
         private int _currentFailedEvaluators = 0;
         private bool _reachedUpdateTaskActiveContext = false;
         private readonly bool _invokeGC;
+        private readonly IList<IActiveContext> _activeContexts = new List<IActiveContext>();
```

Comment:
I do not think this is threadsafe. You will need to use ConcurrentStack or something similar